src/coreclr/jit/importer.cpp

```diff
@@ -10065,8 +10126,11 @@ void Compiler::impImportBlockCode(BasicBlock* block)
 
             case CEE_THROW:
 
-                // Any block with a throw is rarely executed.
-                block->bbSetRunRarely();
+                if (!fgPgoSynthesized)
+                {
+                    // Any block with a throw is rarely executed.
+                    block->bbSetRunRarely();
```

Comment:
Following up on the planned `BasicBlock` interface changes, I imagine this pattern will come up enough that we'll want to move the check into `bbSetRunRarely` (or do some similar cleanup).